eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/ReplicationEndpoint.scala

```diff
@@ -281,11 +284,15 @@ class ReplicationEndpoint(
       val promise = Promise[Unit]()
       val recovery = new Recovery(this)
 
+      def recoveryFailure[U](partialUpdate: Boolean): PartialFunction[Throwable, Future[U]] = {
+        case t => Future.failed(new RecoveryException(t, partialUpdate))
+      }
+
       val recoveryOperation = for {
-        infos <- recovery.readEndpointInfos
-        clocks <- recovery.readEventLogClocks
+        infos <- recovery.readEndpointInfos.recoverWith(recoveryFailure(partialUpdate = false))
+        clocks <- recovery.readEventLogClocks.recoverWith(recoveryFailure(partialUpdate = false))
         links = recovery.recoveryLinks(infos, clocks)
-        _ <- recovery.deleteSnapshots(links)
+        _ <- recovery.deleteSnapshots(links).recoverWith(recoveryFailure(partialUpdate = true))
```

Comment:
Snapshots that cover events that have been lost during a disaster must be removed. This is what  `recovery.deleteSnapshots(links)` is doing. If this operation fails, `partialUpdate` is set to `true` because it might have partially deleted some local snapshots already.